integrations/erc20/hooks/use-deploy.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react'
+
+import { zodResolver } from '@hookform/resolvers/zod'
+import { Signer, ethers } from 'ethers'
+import { useForm } from 'react-hook-form'
+import { useSigner } from 'wagmi'
+import { z } from 'zod'
+
+import { useTokenStorage } from './use-token-storage'
+import { erc20MintableByteCode } from '../abis/erc20-mintable-bytecode'
+import { erc20MintableABI } from '../erc20-wagmi'
+
+export const useDeploy = ({ deployFormSchema }) => {
```

Comment:
You're not defining a type for `deployFormSchema` so it's being implicitly typed as any, so all the instances where you infer the type of it with `z.infer<typeof deployFormSchema>` will not work out as you can see on the screenshots below (that apply for all of the form hooks)
Moving the `formSchema` to each hook makes more sense in my opition, since you're only using it there and it will also sove this typping problem, also your form hooks will need no schema to be passed as an argument. 
<img width="888" alt="image" src="https://github.com/turbo-eth/template-web3-app/assets/18421017/c53afc8b-b11d-4da5-b05a-9f00f5cd456c">
<img width="527" alt="image" src="https://github.com/turbo-eth/template-web3-app/assets/18421017/b7cc615b-45d9-4f74-8c7e-96f2de25f9c3">
<img width="441" alt="image" src="https://github.com/turbo-eth/template-web3-app/assets/18421017/c2fba649-75a8-4159-a045-8d1edc99a7ac">

Reply:
moving the same now